components/Modal/Modal.tsx

```diff
@@ -0,0 +1,42 @@
+import { Modal as WindowModal } from '@mantine/core';
+import React, { useContext } from 'react';
+import { ModalContext } from '../Providers/ModalProvider';
+
+import { InterestZoneProviderContext } from '../Providers/ProviderZone';
+import ModalZone from './ModalZone';
+
+export default function ModalComponent() {
```

Comment:
Please use a more specific name for the component, also avoid using the `Component` suffix (`InterestZoneModal`?)